parser/src/main/kotlin/tools/samt/parser/ASTPrinter.kt

```diff
@@ -0,0 +1,214 @@
+package tools.samt.parser
+
+class ASTPrinter private constructor() {
+    private fun dump(node: Node): String = buildString {
+        append(node.javaClass.simpleName)
+
+        val info = dumpInfo(node)
+        if (info != null) {
+            append(" $info")
+        }
+
+        append(" <${node.location}>")
+        append(System.lineSeparator())
+
+        val childDumps: List<String> = buildList {
+            iterateChildren(node) { child ->
+                add(dump(child))
+            }
+        }
+
+        childDumps.forEachIndexed { childIndex, child ->
+            var firstLine = true
+            child.lineSequence().forEach { line ->
+                if (!line.isEmpty()) {
+                    if (childIndex != childDumps.lastIndex) {
+                        if (firstLine) {
+                            append("├─$line")
+                        } else {
+                            append("│ $line")
+                        }
+                    } else {
+                        if (firstLine) {
+                            append("└─$line")
+                        } else {
+                            append("  $line")
+                        }
+                    }
+
+                    append(System.lineSeparator())
+                }
+
+                firstLine = false
+            }
+        }
+
+        append(System.lineSeparator())
+    }
+
+    private fun dumpInfo(node: Node): String? = when (node) {
+        is RequestResponseOperationNode -> if (node.isAsync) "async" else null
+        is IdentifierNode -> node.name
+        is ImportBundleIdentifierNode -> if (node.isWildcard) "wildcard" else null
+        is IntegerNode -> node.value.toString()
+        is FloatNode -> node.value.toString()
+        is BooleanNode -> node.value.toString()
+        is StringNode -> "\"${node.value}\""
+        else -> null
+    }
+
+    private inline fun iterateChildren(node: Node, body: (Node) -> Unit) {
```

Comment:
Such a parameter would usually be called `block` instead of `body` as far as I know